modules/statistics/php/stats_mri.class.inc

```diff
@@ -110,30 +113,121 @@ class Stats_MRI extends \NDB_Form
                 && in_array($subcat, $subcats)
                 && $this->_inCenter($center, $centres)
             ) {
+                /* FIXME The below is an unholy mess. The isset checks were
+                 * added to do a quick bugfix to prevent PHP Notices from
+                 * cluttering the logs. However this next section of the code
+                 * is very difficult to read and debug.
+                 * We should come up with a better way to do this.
```

Comment:
It's still not amazing, but couldn't we just do

```
if (!isset($tpl_data['data'])) {
     tpl_data['data'] = array();
}
if (!isset($tpl_data['data'][$subproj])) {
     $tpl_data['data'][$subproj] = array(
           'total' => 0,
           $subcat => 0,
      );
}
if (!isset($tpl_data['data'][$subproj][$vl])) {
     $tpl_data['data'][$subproj][$vl] = array(
         $subcat => 0,
         'total' => 0,
     );
}
[...etc]
```

before the `+=` block. It would still be a mess, but it would be cleaner than wraping every single thing in an if isset/else block individually.

Reply:
Yeah that's better. Thanks for the input. I'll refactor this.